code-challenges.js

```diff
@@ -3,34 +3,93 @@
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 
-
-
-
 var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
-// --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
+const secretCode = (string) => {
+    let i = 0
+    let array = string.split("")
+    let newString = "" 
+
+    array.map(value =>{
+        if( value[i] === "a") {
+        newString += 4
+        }
+        else if( value[i] === "e"){
+            newString += 3
+        }else if( value[i] === "i"){
+            newString += 1
+        } else if( value[i] === "o"){
+            newString += 0
+        }  else
+        newString += value[i]
+    } )
+ 
+    return newString
+}
+
+console.log(secretCode(secretCodeWord1));
+console.log(secretCode(secretCodeWord2));
 
 
 
+// --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
 
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
 
 
-
+//functin that takes in an array of words called onlyA
+const onlyA = (array) => {
+//create a new array for the words with an "a"
+    let newArray =[]
+//loop through each string inside the array of words creating a new array for each word
+    for (let i=0; i < array.length; i++){
+        let wordArray = array[i].split("")
+//check each index of each word for an "a"
+            for (let i=0; i < wordArray.length; i++){  
+//if the word contains an "a" push that word to the new array
+                if (wordArray[i].toLowerCase() === "a"){
+                newArray.push(wordArray.join(""))
+                break
+                }
+            }
+    }
+    return newArray
+}
+console.log(onlyA(arrayOfWords));
 
```

Comment:
You got there, I'll give you that! And bonus points for comments. However, this one may have a refactor in its future.